services/web/server/src/simcore_service_webserver/director_v2_abc.py

```diff
@@ -0,0 +1,54 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional, Tuple
+
+from aiohttp import web
+from models_library.projects import ProjectID
+
+CommitID = int
+
+
+class AbstractProjectRunPolicy(ABC):
+    """
+    Since the introduction of meta-projects, not all
+    projects can be executed "as is". E.g. a meta-project
+    with an iterator needs to be converted to a project that
+    replaces the iterator node with a concrete
+    iteration value so it can be run.
+
+    This policy class intends to fill this gap by resolving which
+    runnable/s project/s are associated to a given project (by it's project_uuid)
+
+    For instance, if project_uuid is a standard project, then it returns itself
+    but if project_uuid is a meta-project, then it returns iterations
+
+    NOTE: this abstraction is used to change the functionality of
+    the ``director_v2`` app module while keeping it decoupled from
+    higher level add-ons like the ``meta`` app module
+    """
+
+    # pylint: disable=unused-argument
+
+    @abstractmethod
+    async def get_runnable_projects_ids(
+        self,
+        request: web.Request,
+        project_uuid: ProjectID,
+    ) -> List[ProjectID]:
+        ...
+
+    @abstractmethod
+    async def get_or_create_runnable_projects(
+        self,
+        request: web.Request,
+        project_uuid: ProjectID,
+    ) -> Tuple[List[ProjectID], List[CommitID]]:
```

Comment:
Use a dataclass here as return value. It will be more explicit
```python
@dataclass
class RunnableProjects:
  project_ids: List[ProjectID]
  commit_ids: List[CommitID]
```

Reply:
i tend to slightly disagree here.  I would prefer to define a ``RunnableProject`` and then return ``List[RunnableProject]`` instead

Reply:
there is no question that packaging the data as you suggest increase readibility but I will be forced to unpack them back after the call. for the moment keep this version and look for a more readable yet effective alternative in the next PR